internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -632,17 +653,19 @@ func expandMonitorDiagnosticsSettingsLogs(input []interface{}) []diagnosticsetti
 		}
 		if category != "" {
 			output.Category = utils.String(category)
-		} else {
+		} else if categoryGroup != "" {
 			output.CategoryGroup = utils.String(categoryGroup)
+		} else {
+			return nil, fmt.Errorf("exactly one of `category` or `category_group` must be specified")
```

Comment:
We have the feature `ExactlyOneOf` so this validation can go in the schema and happen earlier in the lifecycle?

Reply:
We can have more than one `log` block, so we cannot use `ExactlyOneOf`.